core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3207,14 +3212,55 @@ private boolean syncMetadata(RpcContext rpcContext, LockedInodePath inodePath,
       }
     }
 
-    if (loadMetadata) {
+    // Update metadata for all the mount points
+    for (String mountPoint : pathsToLoad) {
+      AlluxioURI mountPointUri = new AlluxioURI(mountPoint);
       try {
-        loadMetadataAndJournal(rpcContext, inodePath, LoadMetadataOptions.defaults()
-            .setCreateAncestors(true).setLoadDescendantType(syncDescendantType));
-      } catch (Exception e) {
-        // This may be expected. For example, when creating a new file, the UFS file is not
-        // expected to exist.
-        LOG.debug("Failed to load metadata for path: {}", inodePath.getUri(), e);
+        if (PathUtils.hasPrefix(inodePath.getUri().getPath(), mountPointUri.getPath())) {
+          // one of the mountpoint is above the original inodePath, we start loading from the
+          // original inodePath. It is already locked. so we proceed to load metadata.
+          try {
+            loadMetadataAndJournal(rpcContext, inodePath, LoadMetadataOptions.defaults()
+                .setCreateAncestors(true).setLoadDescendantType(syncDescendantType));
+          } catch (Exception e) {
+            // This may be expected. For example, when creating a new file, the UFS file is not
+            // expected to exist.
+            LOG.debug("Failed to load metadata for path: {}", inodePath.getUri(), e);
+            return false;
+          }
+        } else {
+          try (InodeLockList mountPointLockList =
+                   mInodeTree.lockDescendant(inodePath, lockingScheme.getMode(), mountPointUri)) {
+            if (!mountPointLockList.getInodes().isEmpty()) {
+              List<Inode<?>> inodes = mountPointLockList.getInodes();
+              // Get the last inode in the list which is an inode corresponding to the mountPoint
+              Inode<?> mountPointInode = inodes.get(inodes.size() - 1);
+
+              // Construct an locked inodepath using TempInodePathForDescendant
+              try (TempInodePathForDescendant tempInodePath =
+                       new TempInodePathForDescendant(inodePath)) {
+                tempInodePath.setDescendant(mountPointInode, mountPointUri);
+
+                try {
+                  loadMetadataAndJournal(rpcContext, tempInodePath, LoadMetadataOptions.defaults()
+                      .setCreateAncestors(true).setLoadDescendantType(syncDescendantType));
+                } catch (Exception e) {
+                  LOG.debug("Failed to load metadata for path: {}", mountPointUri, e);
+                }
+                mUfsSyncPathCache.notifySyncedPath(mountPoint);
+              } catch (InvalidPathException e) {
+                LOG.debug("Failed to load metadata for path: {}", mountPointUri, e);
```

Comment:
How can this happen?

Reply:
combined it with another exception handler. This exception should never happen, because we pass a well formatted `inodePath` object that should always have a valid uri into the constructor of `TempInodePathForDescendant`